ethpm_cli/validation.py

```diff
@@ -50,25 +62,63 @@ def validate_solc_output(project_dir: Path) -> None:
 
 
 def validate_install_cli_args(args: Namespace) -> None:
-    validate_target_uri(args.uri)
+    validate_supported_uri(args.uri)
     if args.alias:
         validate_alias(args.alias)
 
     if args.ethpm_dir:
         validate_ethpm_dir(args.ethpm_dir)
 
+    if is_etherscan_uri(args.uri):
+        if not args.package_name or not args.package_version:
+            raise InstallError(
+                "To install an Etherscan verified contract, you must specify both the "
+                "--package-name and --package-version."
+            )
+    else:
+        if args.package_name:
+            raise InstallError(
+                "You cannot redefine the package_name of an existing package. "
+                "Consider aliasing the package instead."
+            )
+
+        if args.package_version:
+            raise InstallError(
+                "You cannot redefine the version of an existing package."
+            )
+
 
 def validate_uninstall_cli_args(args: Namespace) -> None:
     validate_package_name(args.package)
     if args.ethpm_dir:
         validate_ethpm_dir(args.ethpm_dir)
 
 
-def validate_target_uri(uri: URI) -> None:
-    if not is_supported_content_addressed_uri(uri) and not is_valid_registry_uri(uri):
+def validate_package_is_installed(contract_type_id: str, config: "Config") -> None:
+    package, contract_type = contract_type_id.split(":")
+    validate_package_name(package)
+    if not is_package_installed(package, config):
+        raise InstallError(f"{package} is not installed.")
```

Comment:
Better error message might include the names of all of the packages that are installed.